pulsar-broker/src/test/java/org/apache/pulsar/client/api/TopicReaderTest.java

```diff
@@ -86,6 +87,33 @@ public void testSimpleReader() throws Exception {
         producer.close();
     }
 
+    @Test
+    public void testReaderAtSelectedTimestamp() throws Exception {
+         Producer<byte[]> producer = pulsarClient.newProducer().topic("persistent://my-property/my-ns/testReaderAtSelectedTimestamp")
+                .create();
+         for (int i = 0; i < 10; i++) {
+            String message = "my-message-" + i;
+            producer.send(message.getBytes());
+            Thread.sleep(1000);
+        }
+
+        Reader<byte[]> reader = pulsarClient.newReader().topic("persistent://my-property/my-ns/testReaderAtSelectedTimestamp")
+                .startMessageId(3000).create();
+
+        Message<byte[]> msg = null;
+        Set<String> messageSet = Sets.newHashSet();
+        for (int i = 3; i < 10; i++) {
+            msg = reader.readNext(1, TimeUnit.SECONDS);
```

Comment:
why do you have a timeout here? If the CI environment is slow, readNext will fail, no?